File: PortProject.Api/Application/Resources/DTOs/ResourceDto.cs
namespace PortProject.Api.Application.Resources.DTOs;

public class ResourceDto
{
    public string Code { get; set; }
    public string Description { get; set; }
    public string Kind { get; set; }
    public string? AssignedArea { get; set; }
    public string Status { get; set; }
    public int SetupTimeMinutes { get; set; }
    public string OperationalWindowStart { get; set; }
    public string OperationalWindowEnd { get; set; }
    public List<string>? QualificationRequirements { get; set; }

    public int? AverageContainersPerHour { get; set; } // Crane
    public int? ContainersPerTrip { get; set; }        // Truck
    public double? AverageSpeedKmh { get; set; }       // Truck
    public string? OtherUnit { get; set; }             // Other
    public double? OtherGenericValue { get; set; }     // Other
}

File: port-spa-app/src/infrastructure/repositories/resource/resource.dto.ts
// These DTOs (Data Transfer Objects) define the *exact shape*
// of the data we send to and receive from the API.

export interface ResourceCreateDto {
    description: string;
    kind: string;
    assignedArea?: string | null;
    status: string;
    setupTimeMinutes: number;
    operationalWindowStart: string;
    operationalWindowEnd: string;
    qualificationRequirements?: string[] | null;
    averageContainersPerHour?: number | null;
    containersPerTrip?: number | null;
    averageSpeedKmh?: number | null;
    otherUnit?: string | null;
    otherGenericValue?: number | null;
}

export interface ResourceUpdateDto {
    description: string;
    kind: string;
    assignedArea?: string | null;
    status: string;
    setupTimeMinutes: number;
    operationalWindowStart: string;
    operationalWindowEnd: string;
    qualificationRequirements?: string[] | null;
    averageContainersPerHour?: number | null;
    containersPerTrip?: number | null;
    averageSpeedKmh?: number | null;
    otherUnit?: string | null;
    otherGenericValue?: number | null;
}

export interface ResourceUpdateStatusDto {
    NewStatus: string;
}

